File: js/13549.js
// 0-1 bfs
const fs = require('fs');

const [n, k] = fs
  .readFileSync('example.txt')
  .toString()
  .trim()
  .split(' ')
  .map(Number);

const MAX = 100000;

const isVisited = Array(MAX).fill(false);

if (n >= k) {
  console.log(n - k);
} else {
  console.log(bfs());
}

function bfs() {
  const dq = [[n, 0]];
  isVisited[n] = true;

  while (dq.length !== 0) {
    const [loc, cnt] = dq.shift();

    if (loc === k) {
      return cnt;
    }

    for (next of [loc * 2, loc - 1, loc + 1]) {
      if (!isVisited[next] && 0 <= next && next <= MAX) {
        isVisited[next] = true;
        if (next === loc * 2) {
          dq.unshift([next, cnt]);
        } else {
          dq.push([next, cnt + 1]);
        }
      }
    }
  }
}

// queue
// const fs = require('fs');

// const [n, k] = fs
//   .readFileSync('example.txt')
//   .toString()
//   .trim()
//   .split(' ')
//   .map(Number);

// const MAX = 100000;

// const isVisited = Array(MAX).fill(false);

// if (n >= k) {
//   console.log(n - k);
// } else {
//   console.log(bfs());
// }

// function bfs() {
//   const q = [[n, 0]];
//   isVisited[n] = true;

//   let idx = 0;
//   while (q.length !== idx) {
//     const [loc, cnt] = q[idx++];

//     if (loc === k) {
//       return cnt;
//     }

//     for (next of [loc * 2, loc - 1, loc + 1]) {
//       if (!isVisited[next] && 0 <= next && next <= MAX) {
//         isVisited[next] = true;
//         if (next === loc * 2) {
//           q.push([next, cnt]);
//         } else {
//           q.push([next, cnt + 1]);
//         }
//       }
//     }
//   }
// }

File: c++/13549.cpp
// dijkstra
// #include <iostream>
// #include <queue>
// #include <vector>

// #define MAX 100000

// using namespace std;

// bool isVisited[100001];

// int n, k;

// void dijkstra() {
//   priority_queue<pair<int, int>, vector<pair<int, int>>, greater<>> pq;
//   pq.push({0, n});

//   while (!pq.empty()) {
//     int cnt = pq.top().first;
//     int loc = pq.top().second;
//     pq.pop();

//     if (isVisited[loc]) continue;
//     isVisited[loc] = true;

//     if (loc == k) {
//       cout << cnt;
//       return;
//     }

//     if (loc * 2 <= MAX) {
//       pq.push({cnt, loc * 2});
//     }

//     if (loc + 1 <= MAX) {
//       pq.push({cnt + 1, loc + 1});
//     }

//     if (0 <= loc - 1) {
//       pq.push({cnt + 1, loc - 1});
//     }
//   }
// }

// int main() {
//   freopen("example.txt", "r", stdin);
//   cin >> n >> k;

//   dijkstra();
//   return 0;
// }

// 0-1 bfs
#include <deque>
#include <iostream>

#define MAX 100000

using namespace std;

bool isVisited[100001];

int n, k;
int answer;

void bfs() {
  deque<pair<int, int>> dq;
  dq.push_back({0, n});

  while (!dq.empty()) {
    int cnt = dq.front().first;
    int loc = dq.front().second;
    dq.pop_front();

    if (isVisited[loc]) continue;
    isVisited[loc] = true;

    if (loc == k) {
      cout << cnt;
      return;
    }

    if (loc * 2 <= MAX) {
      dq.push_front({cnt, loc * 2});
    }

    if (loc + 1 <= MAX) {
      dq.push_back({cnt + 1, loc + 1});
    }

    if (0 <= loc - 1) {
      dq.push_back({cnt + 1, loc - 1});
    }
  }
}

int main() {
  freopen("example.txt", "r", stdin);
  cin >> n >> k;

  bfs();
  return 0;
}